### PR TITLE
CF Firecrackers Internal Solution

### PR DESCRIPTION
**Pull request by jessechoe10**

### CODE REVIEW

File: solutions/cf-1468D.mdx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+### C++ Implementation
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+ll cooridor_size, num_of_firecrackers, hooligan_location, guard_location, maxFirecrackers, timeBeforeCaught;
```
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:43]: please put each declaration on a single line

File: solutions/cf-1468D.mdx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+bool works(vl explodingTimes, ll firecrackers_exploded){
+	int current_time = 1;
+	for(int i = firecrackers_exploded - 1; i >= 0; i--){
```
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:47]: add whitespace into these control statements wont you

File: solutions/cf-1468D.mdx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+		if(current_time + explodingTimes[i] > timeBeforeCaught){
+			return false;
+		} else {
+			++current_time;
```
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:51]: i'd honestly prefer `current_time++` but that's just me

File: solutions/cf-1468D.mdx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		} else {
+			right = mid - 1;
+		}
+	}
+	cout << maxExploded << endl;
+}
+
+int main(){
+	int t; cin >> t;
+	for(int i = 1; i <= t; i++){
```
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:87]:
```suggestion
	for (int i = 0; i < t; i++) {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
can't be bothered to look at the editorial lmao

File: solutions/cf-1468D.mdx (hunk 3 above)
Comment (jessechoe10) [reply solutions/cf-1468D.mdx:51]: yeah it doesn't really matter.

**Comment by SansPapyrus683** [comment]
you checked the line lengths and stuff right

**Comment by jessechoe10** [comment]
> you checked the line lengths and stuff right

no not really lmao

File: solutions/cf-1468D.mdx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+			return false;
+		} else {
+			++current_time;
+		}
+	}
+	return true;
+}
+
+void solve() {
+	cin >> cooridor_size >> num_of_firecrackers >> hooligan_location >> guard_location;
```
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:63]:
```suggestion
	cin >> cooridor_size >> num_of_firecrackers
		>> hooligan_location >> guard_location;
```

File: solutions/cf-1468D.mdx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+	vl exploding_times(num_of_firecrackers);
+	for (int i = 0; i < num_of_firecrackers; i++) {
+		cin >> exploding_times[i];
+	}
+	sort(exploding_times.begin(), exploding_times.end());
+	maxFirecrackers = min(num_of_firecrackers, abs(hooligan_location - guard_location) - 1);
```
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:69]:
```suggestion
	maxFirecrackers = min(
		num_of_firecrackers,
		abs(hooligan_location - guard_location) - 1
	);
```

File: solutions/cf-1468D.mdx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+### C++ Implementation
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+ll cooridor_size;
```
Comment (maggie-j-liu) [inline solutions/cf-1468D.mdx:43]: typo? should be corridor

File: solutions/cf-1468D.mdx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+			maxExploded = mid;
+			left = mid + 1;
+		} else {
+			right = mid - 1;
+		}
+	}
+	cout << maxExploded << endl;
+}
+
+int main(){
```
Comment (maggie-j-liu) [inline solutions/cf-1468D.mdx:94]:
```suggestion
int main() {
```

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]
lol the variable names are inconsistent again (could you try to make them consistent next time? not a big deal so you don't have to change them)

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/cf-1468D.mdx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+ll num_of_firecrackers;
+ll hooligan_location; 
+ll guard_location; 
+ll max_firecrackers;
+ll time_before_caught;
+
+bool works(vl exploding_times, ll firecrackers_exploded) {
+	int current_time = 1;
+	for (int i = firecrackers_exploded - 1; i >= 0; i--) {
+		if(current_time + exploding_times[i] > time_before_caught){
```
Comment (nwatx) [inline solutions/cf-1468D.mdx:53]:
```suggestion
		if(current_time + exploding_times[i] > time_before_caught) {
```

File: solutions/cf-1468D.mdx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+id: cf-1468D
+source: CF
+title: Firecrackers
+author: Jesse Choe
+---
+
+[Official Editorial](https://assets.codeforces.com/rounds/1468/tutorials.pdf)
+
+We can initially attempt this problem by figuring out which firecrackers should explode. Obviously, we should attempt to explode the firecrackers with a minimal
+detonation time to increase the number of firecrackers exploded before eventually being caught. We can also observe that the firecrackers with a longer detonation
```
Comment (nwatx) [inline solutions/cf-1468D.mdx:11]:
```suggestion
detonation time to increase the number of firecrackers exploded before eventually being caught. Also, observe that the firecrackers with a longer detonation 
```

File: solutions/cf-1468D.mdx [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+author: Jesse Choe
+---
+
+[Official Editorial](https://assets.codeforces.com/rounds/1468/tutorials.pdf)
+
+We can initially attempt this problem by figuring out which firecrackers should explode. Obviously, we should attempt to explode the firecrackers with a minimal
+detonation time to increase the number of firecrackers exploded before eventually being caught. We can also observe that the firecrackers with a longer detonation 
+time should be dropped first.
+
+To find the maximum number of firecrackers before the hooligan gets caught by the guard, we can sort the detonation times in increasing order and figure out the
```
Comment (nwatx) [inline solutions/cf-1468D.mdx:14]:
```suggestion
To find the maximum number of firecrackers before the hooligan gets caught by the guard, sort the detonation times in increasing order and figure out the
```

File: solutions/cf-1468D.mdx [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+ll corridor_size;
+ll num_of_firecrackers;
+ll hooligan_location; 
+ll guard_location; 
+ll max_firecrackers;
+ll time_before_caught;
```
Comment (nwatx) [inline solutions/cf-1468D.mdx:48]: declare on one / 2 lines?

File: solutions/cf-1468D.mdx [hunk 13]
```diff
@@ -0,0 +1,104 @@
+---
+id: cf-1468D
+source: CF
+title: Firecrackers
+author: Jesse Choe
+---
+
+[Official Editorial](https://assets.codeforces.com/rounds/1468/tutorials.pdf)
```
Comment (nwatx) [inline solutions/cf-1468D.mdx:8]: would it be nicer to link to the [blog](https://codeforces.com/blog/entry/85951) directly?

**Review by nwatx — CHANGES_REQUESTED** [review]

File: solutions/cf-1468D.mdx (hunk 11 above)
Comment (SansPapyrus683) [reply solutions/cf-1468D.mdx:14]: don't feel like this one's that necessary

File: solutions/cf-1468D.mdx [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+<CPPSection>
+
+### C++ Implementation
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
```
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:41]: iirc `vl` wasn't part of the template, so maybe remove this `using`?

File: solutions/cf-1468D.mdx (hunk 12 above)
Comment (SansPapyrus683) [reply solutions/cf-1468D.mdx:48]: no it's best to keep each variable on its own line

File: solutions/cf-1468D.mdx (hunk 9 above)
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:53]: idk maybe it'd be pog to add some comments here

File: solutions/cf-1468D.mdx [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+	vl exploding_times(num_of_firecrackers);
+	for (int i = 0; i < num_of_firecrackers; i++) {
+		cin >> exploding_times[i];
+	}
+	sort(exploding_times.begin(), exploding_times.end());
+	max_firecrackers = min(
+		num_of_firecrackers,
+		abs(hooligan_location - guard_location) - 1
+	);
+	if (hooligan_location < guard_location) {
```
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:74]: like here as well maybe

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cf-1468D.mdx (hunk 12 above)
Comment (nwatx) [reply solutions/cf-1468D.mdx:48]: Fair enough

File: solutions/cf-1468D.mdx [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+	vl exploding_times(num_of_firecrackers);
+	for (int i = 0; i < num_of_firecrackers; i++) {
+		cin >> exploding_times[i];
+	}
+	sort(exploding_times.begin(), exploding_times.end());
+	max_firecrackers = min(
+		num_of_firecrackers,
+		abs(hooligan_location - guard_location) - 1
+	);
+	// Compute the maximum time before the guard catches the hooligan given their initial positions.
```
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:75]: comments seem a bit too long
```cpp
/*
 * maybe split it up and use multiline comments
 * like so
 */
```

File: solutions/cf-1468D.mdx [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+
+After computing the amount of time before getting caught, we can simulate whether a particular firecracker will explode in time.
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+### C++ Implementation
```
Comment (HarryWangATX) [inline solutions/cf-1468D.mdx:34]:
```suggestion
## C++ Implementation
```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: solutions/cf-1468D.mdx [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+	}
+	sort(exploding_times.begin(), exploding_times.end());
+	max_firecrackers = min(
+		num_of_firecrackers,
+		abs(hooligan_location - guard_location) - 1
+	);
+	/* 
+	 *Compute the maximum time before the guard catches the hooligan given 
+	 *their initial positions.
+	 */
```
Comment (SansPapyrus683) [inline solutions/cf-1468D.mdx:81]: uh these comments are styled a bit weird, remember to put a space between the asterisk and the text

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/cf-1468D.mdx (hunk 18 above)
Comment (darren-yao) [reply solutions/cf-1468D.mdx:79]: Fixed.

**Review by darren-yao — APPROVED** [review]

**Pull request merged**